src/lightning/fabric/loggers/csv_logs.py

```diff
@@ -220,12 +221,35 @@ def save(self) -> None:
         if not self.metrics:
             return
 
-        last_m = {}
-        for m in self.metrics:
-            last_m.update(m)
-        metrics_keys = list(last_m.keys())
+        current_keys = _get_keys_from_metrics(self.metrics)
+        new_keys = current_keys - set(self.metrics_keys)
+        self.metrics_keys.extend(new_keys)
```

Comment:
Would you mind moving this to a method? This way the Lit-GPT version can just override this piece: https://github.com/Lightning-AI/lit-gpt/blob/main/lit_gpt/utils.py#L436-L437

Reply:
That's interesting, I forgot that we have this. I can make it a method yes, but the question is should we just do this anyway in the CSVLogger?

Reply:
It might be too opinionated. They could also be merged by "epoch". Or neither "epoch" or "step" might be logged. I'm not sure.

The original code is mine from https://github.com/jpuigcerver/PyLaia/blob/master/laia/loggers/epoch_csv_logger.py where I used to group epoch-level metrics